### ISSUE TITLE
🐛 Unexpected unused export when namespace used in array

### ISSUE THREAD

**Issue by mlahargou** [webpro-nl/knip#844]
### Prerequisites

- [X] I'm using the latest version
- [X] I've read the relevant [documentation](https://knip.dev)
- [X] I've searched for [existing issues](https://github.com/webpro-nl/knip/issues?q=is%3Aissue)
- [X] I've checked the [list of known issues](https://knip.dev/reference/known-issues)
- [X] I've read the [issue reproduction guide](https://knip.dev/guides/issue-reproduction)

### Reproduction url

https://stackblitz.com/edit/stackblitz-starters-evsylk?file=index.js

### Reproduction access

- [X] I've made sure the reproduction is publicly accessible

### Description of the issue

I have a use case where I import a namespace. I then add that namespace to an array and pass it to a function. Knip is marking the namespace exports as unused. I've ensured that `nsExports` is not included.

Repro:

```js
// namespace.js
export const foo = 'foo';
export const bar = 'bar';
```

```js
// index.js
import * as NS from './namespace.js';

// This does throw errors
console.log([NS]);

// This doesn't throw errors
// console.log(NS);
```

#### Output:
```
> npm exec knip
Unused exports (2)
foo  NS  unknown  namespace.js:1:14
bar  NS  unknown  namespace.js:2:14
```

**Comment by webpro** [comment]
Thanks for the report @mlahargou, fix will be in the next one!

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.37.2. See [Release 5.37.2](https://github.com/webpro-nl/knip/releases/tag/5.37.2) for release notes._

_Using Knip in a commercial project? Please consider [becoming a sponsor](https://knip.dev/sponsors)._
